src/streaming/controllers/PlaybackController.js

```diff
@@ -104,10 +101,28 @@ function PlaybackController() {
         }
     }
 
-    function onPeriodSwitchStarted(e) {
-        if (!isDynamic && e.fromStreamInfo && earliestTime[e.fromStreamInfo.id] !== undefined) {
-            delete bufferedRange[e.fromStreamInfo.id];
-            delete earliestTime[e.fromStreamInfo.id];
+    function onStreamInitialized(e) {
+        // Seamless period switch
+        if (streamSwitch && isNaN(streamSeekTime)) return;
+
+        // Seek new stream in priority order:
+        // - at seek time (streamSeekTime) that required period switch
+        // - at start time provided in URI parameters
+        // - at stream/period start time (for static streams) or live start time (for dynamic streams)
+        let startTime = streamSeekTime;
+        if (isNaN(startTime)) {
+            startTime = isDynamic ? e.liveStartTime : e.streamInfo.start;
+            startTime = Math.max(startTime, getStartTimeFromUri());
```

Comment:
Following up the discussion we had on Slack and using #t during startup: Using #t for live will not work with dynamic manifests as the live edge will always have a higher value than the #t. I think it should be Math.min but we need to consider the DVR window at this point. If #t is out of the DVR window we can either ignore it or adjust its value.

Reply:
Addition: For Math.min for live, Math.max for VoD

Reply:
getStartTimeFromUri() should return NaN instead of -1

Reply:
No, getStartTimeFromUri() should not return NaN
Math.max and Math.min are expecting each parameter to ba a valid number
Math.max(x, NaN) and Math.min(x, NaN) return NaN

But I do modify acording your remarks and take advantage of this PR to make code conformed to MPD anchors spec